### ISSUE TITLE
Cooldown + Wrong skin

### ISSUE THREAD

**Issue by Bodyash** [games647/ChangeSkin#28]
/skin Nickname
/skin AnotherNickname - error (not so fast, must wait, but in config cooldown is set to 0)

And 1 more - MickeyPickles nickname. My Friend changed skin at Mojang website (http://mcskinsearch.com/skin/MickeyPickles)
But displaying an old one skin data.
![2016-06-28_03 23 11](https://cloud.githubusercontent.com/assets/2454569/16399946/ad0a39a6-3cdf-11e6-9ad8-ea3b1c8d6d41.png)
Restart Server and /skin MickeyPickles doesnt update skin. (Latest ChangeSkin build)

**Comment by games647** [comment]
>And 1 more - MickeyPickles nickname. My Friend changed skin at Mojang website (http://mcskinsearch.com/skin/MickeyPickles)
But displaying an old one skin data.

Therefore the command /skinupdate exists. I could also make it run automatically if you want to.

I'll look into the other issues. 

>(Latest ChangeSkin build)

The latest version from Github or Spigot?

**Comment by Bodyash** [comment]
Github releases page

**Comment by Augustin37** [comment]
Could you also add the "/skinupdate player" command to update the skin from the console? That would be cool :) Thank you!

**Comment by games647** [comment]
@Bodyash Big sorry for the big delay. I was too busy. It's fixed now

**Comment by games647** [comment]
@Augustin37 With or without a message for the receiver (not the invoker)?

**Comment by Augustin37** [comment]
With a message for the receiver that would be great! :)

**Comment by games647** [comment]
@Augustin37 I'll add it for 1.9

**Comment by Augustin37** [comment]
Thank you :D

**Comment by Bodyash** [comment]
Version from Spigotmc with permissions problem

**Comment by games647** [comment]
@Bodyash Could you describe it more? What's the problem? Do you use BungeeCord? Do changed any setting? Which permission message do you see?

**Comment by Bodyash** [comment]
Only Spigot 1.10.2, no bungee.

`# Specific skin permissions are activated and the user doesn't have the permission for the supplied skin
no-permission: "&4У тебя нет прав на смену скина :с"`

/skin Bodyash (player with changeskin.command.setskin permission via PermissionsEX)

![default](https://cloud.githubusercontent.com/assets/2454569/16485242/858e334e-3ec6-11e6-84ce-642d4a036365.PNG)

First message - queue-name-resolve
Second - uuid-resolved

**Comment by games647** [comment]
Do you have skin-permissions activated?

**Comment by Bodyash** [comment]
Yeap, true in config

**Comment by games647** [comment]
Do you give yourself the changeskin.skin.whitelist.<uuid> ?

<uuid> replaced with the target uuid

**Comment by Bodyash** [comment]
no, i`m op
But players must have that permission?
For example i can add changeskin.skin.whitelist.* ?

**Comment by games647** [comment]
Indeed. Therefore this option exists. 

Checking the permission changeskin.command.setskin is not related to this option. They are always checked.

**Comment by Bodyash** [comment]
Okay, i will try. Can u list all current permissions at https://www.spigotmc.org/resources/changeskin.21469/ and update info at dev.bukkit page?

**Comment by games647** [comment]
Of course. Until I finished it, you can use the list on the project ReadMe. https://github.com/games647/ChangeSkin#permissions

Just to keep in mind the whitelist and blacklist permissions are only used if the config options is enabled
